drivers/docker/stats.go

```diff
@@ -141,13 +136,20 @@ func (h *taskHandle) collectStats(ctx context.Context, destCh *usageSender, inte
 			retry++
 			continue
 		}
+		defer statsReader.Body.Close()
+
+		var stats containerapi.Stats
+		binary.Read(statsReader.Body, binary.LittleEndian, &stats)
```

Comment:
Should we catch decoding errors here (but not things like `io.EOF`)? I feel like if we collected garbage we probably don't want to send that upstream to throw errors because of empty fields in the decoded object.

Reply:
Yes. I actually ended up revamping this because that implementation was buggy. See [ee27028](https://github.com/hashicorp/nomad/pull/23966/commits/ee2702837362adf4ec9a222e5cf9cfe6d30888d2) for improvements.